### DISCUSSION TITLE
CorruptComponentPayloadException when query string change on back button in combination with computed properties

### DISCUSSION THREAD

**Discussion by waterbuckit** [livewire/livewire#4716] [7. Bugs (archived)]
Livewire Version: v2.10.4 

Apologies for the rather in-concise title, and feel free to shoot me down for using things incorrectly, but I have a component ``Test.php``:
```php
class Test extends Component
{
    use WithPagination;

    public $show = true;
    public $queryString = ['show'];

    public function action(){
        $this->show = !$this->show;
    }
    
    public function getModelARowsProperty(){
        return ModelA::paginate(3, ['*'], 'modelAPagination'); 
    }

    public function getModelBRowsProperty(){
        return ModelB::paginate(3, ['*'], 'modelBPagination'); 
    }

    public function render()
    {
        return view('livewire.test');
    }
}
```
and a blade view as such:
```blade
<div>

    @if($show)
    <ul>
        @foreach($this->modelARows as $row)
            <li wire:key="model-a-{{$row->id}}">{{$row->id}}</li>
        @endforeach
    </ul>
    <div>
        {{$this->modelARows->links()}}
    </div>
    @else
    <ul>
        @foreach($this->modelBRows as $row)
            <li wire:key="model-b-{{$row->id}}">{{$row->id}}</li>
        @endforeach
    </ul>
    <div>
        {{$this->modelBRows->links()}}
    </div>
    @endif

    <button wire:click="action">Test</button>

</div>
```
If I click the button, which triggers the action, "show" is toggled. This in turn changes the query string, the issue is that when I press the back navigation in the browser, it returns to the original state of "show" and clicking the button again then throws the `CorruptComponentPayloadException`.

I may be using this incorrectly, and this issue may stem potentially from using computed properties the way I am, but I don't see why it should be a problem? (this example is derived from a much larger interface which has been implemented in a similar way)

Is this expected behaviour?

Thanks for the awesome package.

**Comment by joshhanley** [comment]
@waterbuckit I've just tested your code in my test repo but it all worked happily for me. Any chance you could create a small example repo which reproduces the issue with as little code as possible so we can investigate it?

Thanks!

**Reply by waterbuckit** [reply]
Will make you a repo later - is that LivewirePlayground repo you have good to use?

**Reply by joshhanley** [reply]
@waterbuckit thanks! Yeah, it just has a component called main on the home route which makes it easy to dump Livewire experiments into.

I've pushed the branch of your code I tested if you wanted to have a look https://github.com/joshhanley/LivewirePlayground/tree/back-button-corruption-test

**Reply by waterbuckit** [reply]
@joshhanley Cheers for setting that up! So I forked the repo and switched to that branch - I'm still getting the exception, can you confirm you followed these steps to reproduce:

1. Click "Test"
2. Navigate back in the browser
3. Click "Test" again

It's at this point the exception is thrown.

**Reply by joshhanley** [reply]
@waterbuckit thanks for the steps, yeah I managed to trigger the error too.

So looking into it, it's because the passed data changes, hence why it is failing the checksum.

This is the data before I click test (the second time)

<img width="401" alt="image" src="https://user-images.githubusercontent.com/882837/155924730-e5647fe2-4ed4-41c3-9eca-5b1af0077666.png">

And this is data when the check is happening as part of clicking test (the second time)

<img width="385" alt="image" src="https://user-images.githubusercontent.com/882837/155924811-6c5ccf3f-0d0c-4d97-a549-b698c9e047f4.png">

You can see in the first screenshot under the paginators property is different to the second screenshot.

So I suspect due to the computed properties it's not loading the model b paginator properly in the paginators stack the second time.

It's likely to be a bug with how history state is managed but if it is, it's not something simple to fix. I would try and see if you can work around it for the moment.

Hope this helps!

**Reply by waterbuckit** [reply]
Okie doke, thanks for the reply - I figured it had to be something to do with the computed properties in combination with pagination and querystring all at once.
My solution will probably end up being conditionally passing view props - just wanted to sincerely avoid that as I've got a complex fullpage component with many paginations (lots of tabs with different tables etc) and having to condtionally pass pagination props rather than just define them in the component as computed props and declare them in my blade is a nightmare 😭😂

Thanks again though.

Is there any chance you will attempt to fix this?

**Reply by joshhanley** [reply]
@waterbuckit yeah fair enough. You might be able to work around it by manually adding data to the `$paginators` property on mount so it is there for all requests (although I wouldn't generally recommend manually touching it, it might be worth experimenting).

Honestly, probably not. At least not at this stage. If it were a simple thing to investigate and fix, I'd probably try and have a look at it. 

But V3 of Livewire will involve a re-write of the front end and as such the history state code, so we may be able to either fix it then, or it should at least make it easier for fixing in the future.

If you feel like digging in and seeing if you can fix it, PR's are welcome!
